packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -427,6 +431,31 @@ export class StripeWebhookHandler extends StripeHandler {
     await this.updateCustomer(uid, email);
   }
 
+  /**
+   * Validate product metadata on update
+   */
+  async handleProductUpdate(request: AuthRequest, event: Stripe.Event) {
+    const product = event.data.object as Stripe.Source;
+    const result = subscriptionProductMetadataValidator.validate(
+      product.metadata
+    );
```

Comment:
The same issue with validating only the product metadata applies here.  On a product update, if the metadata have been updated (you can check the event's "previous_attributes" property for a "metadata" key), then we should validate all the plans for the product.